yapapi/rest/market.py

```diff
@@ -191,8 +194,10 @@ async def events(self) -> AsyncIterator[OfferProposal]:
         """Yield counter-proposals based on the incoming, matching Offers."""
         while self._open:
 
+            proposals = []
             try:
-                proposals = await self._api.collect_offers(self._id, timeout=10, max_events=10)
+                async with SuppressedExceptions(is_recoverable_exception):
+                    proposals = await self._api.collect_offers(self._id, timeout=5, max_events=10)
```

Comment:
Wrapping this call in `SuppressedExceptions(is_recoverable_exception)` would ignore timeout errors and `aiohttp.ServerDisconnectedError`.